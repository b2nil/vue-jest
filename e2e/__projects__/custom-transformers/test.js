import { mount } from '@vue/test-utils'
import NodeTransform from './components/NodeTransform.vue'

test('should honor sfc template compile options provided in globals', () => {
  const warn = jest.spyOn(console, 'warn').mockImplementation(jest.fn())

  const wrapper = mount(NodeTransform)
  expect(wrapper.element).toMatchSnapshot()

  const viewEl = wrapper.find('.light')
  const scrollViewEl = wrapper.find('.dark')

  expect(viewEl.element.tagName).toEqual('VIEW')
  expect(scrollViewEl.element.tagName).toEqual('SCROLL-VIEW')
  expect(wrapper.findComponent('.light').exists()).toBeFalsy()
  expect(wrapper.findComponent('.dark').exists()).toBeFalsy()

  expect(wrapper.findComponent('.comp').exists()).toBeTruthy()

  expect(wrapper.findComponent('.unkown').exists()).toBeFalsy()

  expect(warn.mock.calls[0][0]).toEqual(
    '[Vue warn]: Failed to resolve component: unknown-tag'
  )
})

// TODO: Figure this out with Vue 3. `$style` no longer exists.
test.todo('processes SCSS using user specified post transforms')

test.todo('processes SCSS using user specified pre transforms')
