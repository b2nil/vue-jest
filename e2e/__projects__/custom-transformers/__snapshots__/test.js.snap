// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should honor sfc template compile options provided in globals 1`] = `
<div>
  <view
    class="light"
  >
    should be treated as native tag
  </view>
  <div
    class="comp"
  >
    
    should resolve this tag as component
    
  </div>
  <scroll-view
    class="dark"
  >
    should be treated as native tag
  </scroll-view>
  <unknown-tag
    class="unkown"
  >
    should fail to resolve this tag 
  </unknown-tag>
</div>
`;
